src/bbsecondary.c

```diff
@@ -184,8 +184,10 @@ bool start_secondary(bool need_secondary) {
   //check if X is available, for maximum 10 seconds.
   time_t xtimer = time(0);
   Display * xdisp = 0;
+  char unix_x_display[5 + sizeof(bb_config.x_display)] = "unix/";
+  strcat(unix_x_display, bb_config.x_display);
```

Comment:
`sizeof(bb_config.c_display)` is not correct, it gives the size of a pointer. Also `strcat` should not be used due to the risk of a buffer overrun.

What about:
```c
char unix_x_display[16];
snprintf(unix_x_display, sizeof(unix_x_display), "unix/%s", bb_config.x_display);
```

Alternatively, you could set `g_strdup_printf("unix/%s", bb_config.x_display)`, but that needs to be freed afterwards.